lms/djangoapps/bookmarks/tests/test_models.py

```diff
@@ -0,0 +1,100 @@
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+
+class BookmarkModelsTest(ModuleStoreTestCase):
+    """
+    Test the Bookmark model.
+    """
+    def setUp(self):
+        super(BookmarkModelsTest, self).setUp()
+
+        self.user = UserFactory.create(password="test")
+
+        self.course = CourseFactory.create(display_name='An Introduction to API Testing')
+        self.course_id = unicode(self.course.id)
+
+        chapter = ItemFactory.create(
+            parent_location=self.course.location, category='chapter', display_name="Week 1"
+        )
+        sequential = ItemFactory.create(
+            parent_location=chapter.location, category='sequential', display_name="Lesson 1"
+        )
+        self.vertical = ItemFactory.create(
+            parent_location=sequential.location, category='vertical', display_name='Subsection 1'
+        )
+        # This list shows the path, that we store in db.
+        self.breadcrumb = [
+            {"display_name": chapter.display_name, "usage_id": unicode(chapter.location)},
+            {"display_name": sequential.display_name, "usage_id": unicode(sequential.location)}
+        ]
+        # Dict to make the bookmarks
+        self.bookmark_data = {
+            "usage_key": self.vertical.location,
+            "course_key": self.course.id,
+            "user": self.user
+        }
+
+    def assert_valid_bookmark(self, bookmark_object, bookmark_data):
+        """
+        Determines if the given data matches the specified bookmark.
+        """
+        self.assertEqual(bookmark_object.usage_key, bookmark_data['usage_key'])
+        self.assertEqual(bookmark_object.course_key, bookmark_data['course_key'])
+        self.assertIsNotNone(bookmark_object.created)
+        self.assertEqual(bookmark_object.user, self.user)
+
+        self.assertEqual(bookmark_object.path, self.breadcrumb)
+        self.assertEqual(bookmark_object.display_name, self.vertical.display_name)
+
+    def test_create_bookmark_successfully(self):
+        """
+        Tests creation of bookmark.
+        """
+        bookmark_object = Bookmark.create(bookmarks_data=self.bookmark_data)
+        self.assertIsNotNone(bookmark_object)
+        self.assert_valid_bookmark(bookmark_object, self.bookmark_data)
+
+    def test_create_raises_exception_for_unreachable_block(self):
+        """
+        Tests creation of bookmark for unreachable block.
+        """
+        temp_data = self.bookmark_data
+        temp_data['usage_key'] = UsageKey.from_string('i4x://arbi/100/html/340ef1771a0940')
+        with self.assertRaises(ItemNotFoundError):
+            Bookmark.create(bookmarks_data=temp_data)
+
+    def test_create_bookmark_with_given_block(self):
+        """
+        Tests creation of bookmark with block given.
+        """
+        bookmark_object = Bookmark.create(bookmarks_data=self.bookmark_data, block=self.vertical)
```

Comment:
Can you create a different vertical or remove fields from bookmark_data? Right now, this looks identical to "test_create_bookmark_successfully", and it would seem that it could pass if the code just ignores the specified "block".